### PR TITLE
[android] Fix crash in TTS settings

### LINKED ISSUE #4728: [android] A crash in SettingsPrefFragment.initLangInfoLink()
Android 7.0 (SDK 24)

```
Exception android.content.ActivityNotFoundException:
  at android.app.Instrumentation.checkStartActivityResult (Instrumentation.java:1809)
  at android.app.Instrumentation.execStartActivity (Instrumentation.java:1523)
  at android.app.Activity.startActivityForResult (Activity.java:4226)
  at androidx.activity.ComponentActivity.startActivityForResult (ComponentActivity.java)
  at android.app.Activity.startActivityForResult (Activity.java:4185)
  at androidx.activity.ComponentActivity.startActivityForResult (ComponentActivity.java)
  at android.app.Activity.startActivity (Activity.java:4509)
  at android.app.Activity.startActivity (Activity.java:4477)
  at app.organicmaps.settings.SettingsPrefsFragment.lambda$initLangInfoLink$0 (SettingsPrefsFragment.java)
  at app.organicmaps.settings.SettingsPrefsFragment.$r8$lambda$0-w8kLs9DdQH2DHLpCLSdFSQ3T0 (SettingsPrefsFragment.java)
  at app.organicmaps.settings.SettingsPrefsFragment$$InternalSyntheticLambda$1$b04810de1e44b7fb29d08a50076248026e289747182bc745258b395b8546407c$0.onPreferenceClick (SettingsPrefsFragment.java)
  at androidx.preference.Preference.performClick (Preference.java)
  at androidx.preference.Preference.performClick (Preference.java)
  at androidx.preference.Preference$1.onClick (Preference.java)
  at android.view.View.performClick (View.java:5675)
  at android.view.View$PerformClick.run (View.java:22641)
  at android.os.Handler.handleCallback (Handler.java:836)
  at android.os.Handler.dispatchMessage (Handler.java:103)
  at android.os.Looper.loop (Looper.java:203)
  at android.app.ActivityThread.main (ActivityThread.java:6251)
  at java.lang.reflect.Method.invoke (Method.java)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1063)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:924)

```

https://play.google.com/console/u/0/developers/8084273541548442467/app/4972992444534608608/vitals/crashes/94ff33abb6267fd7ba90572a4265e943/details?days=28&versionCode=23030505%2C23030203

https://github.com/organicmaps/organicmaps/blob/b1269be1a4ab5451b6a04c27f46f3b39a950ef46/android/src/app/organicmaps/settings/SettingsPrefsFragment.java#L358-L361

### PR DESCRIPTION
**Pull request by Jean-BaptisteC**
Fixes #4728
Tested on Pixel 6 - Android 14 (I don't have tested without browser)

### CODE REVIEW

**Review by biodranik — APPROVED** [review]
Thanks, good catch. Do we have other similar places in the code?

It would be helpful to print a toast like "Web browser is not available" when openUri fails.

**Comment by Jean-BaptisteC** [comment]
Yes, I have checked, it's only case where we do not use utils method.
It's make sense to use OpenUrl instead OpenUri.
Do you think it's necessary to use translations for text in toast?
Because device with no browser is rare maybe english text is enough.

**Comment by biodranik** [comment]
Those few people who hit this issue will be grateful for the readable translation :)

```
    en = Web browser is not available
    af = Webblaaier is nie beskikbaar nie
    ar = متصفح الويب غير متوفر
    be = Вэб-браўзер недаступны
    bg = Уеб браузърът не е наличен
    ca = El navegador web no està disponible
    cs = Webový prohlížeč není k dispozici
    da = Webbrowser er ikke tilgængelig
    de = Webbrowser ist nicht verfügbar
    el = Το πρόγραμμα περιήγησης στο Web δεν είναι διαθέσιμο
    es = El navegador web no está disponible
    et = Veebibrauser ei ole saadaval
    eu = Web arakatzailea ez dago erabilgarri
    fa = مرورگر وب در دسترس نیست
    fi = Verkkoselain ei ole käytettävissä
    fr = Le navigateur Web n'est pas disponible
    he = דפדפן אינטרנט אינו זמין
    hi = वेब ब्राउज़र उपलब्ध नहीं है
    hu = A webböngésző nem elérhető
    id = Browser web tidak tersedia
    it = Il browser web non è disponibile
    ja = ウェブブラウザが利用できない
    ko = 웹 브라우저를 사용할 수 없습니다.
    lt = Interneto naršyklė nepasiekiama
    mr = वेब ब्राउझर उपलब्ध नाही
    nb = Nettleser er ikke tilgjengelig
    nl = Webbrowser is niet beschikbaar
    pl = Przeglądarka internetowa jest niedostępna
    pt = O navegador da Web não está disponível
    ro = Browserul web nu este disponibil
    ru = Веб-браузер недоступен
    sk = Webový prehliadač nie je k dispozícii
    sv = Webbläsaren är inte tillgänglig
    sw = Kivinjari cha wavuti hakipatikani
    th = เว็บเบราว์เซอร์ไม่พร้อมใช้งาน
    tr = Web tarayıcısı mevcut değil
    uk = Веб-браузер недоступний
    vi = Trình duyệt web không khả dụng
    zh-Hans = 网络浏览器不可用
    zh-Hant = 網路瀏覽器不可用
```

**Comment by biodranik** [comment]
Btw, we also render bookmarks description using WebView. It would be great to display there a similar toast if system webview/browser is disabled or not available.

Is it possible to disable it for testing like this? https://www.digitaltrends.com/mobile/what-is-android-system-webview/

**Comment by Jean-BaptisteC** [comment]
If I disable webview in my device, app crash when I try to open FAQ or wikipedia in place page

**Comment by biodranik** [comment]
Great! It should not crash but show a toast instead.

**Comment by Jean-BaptisteC** [comment]
- I have added Toast in openUrl method (I have planned to regenerate translations when PR is finish before merge).
- I have created new openWebview method to catch ActivityNotFoundException, (I need to check if I can use this to place page wikipedia or webview in bookmarks)

File: android/app/src/main/java/app/organicmaps/util/Utils.java [hunk 1]
```diff
@@ -774,4 +775,15 @@ public static PackageInfo getPackageInfo(@NonNull PackageManager manager, @NonNu
       return getPackageInfoOld(manager, packageName, flags);
     return manager.getPackageInfo(packageName, PackageManager.PackageInfoFlags.of(flags));
   }
+
+  public static void openWebview(@NonNull Context context, @NonNull Intent intent){
```
Comment (biodranik) [inline android/app/src/main/java/app/organicmaps/util/Utils.java:779]: It's not a good idea to merge methods that are not used yet (and not confirmed that they are working yet). Let's split tasks and move incrementally:
1. Finish this PR with openUrl and translations
2. Make an issue for a separate task to handle missing webview crashes
3. After that, a fix can be tested/developed for webviews.

**Review by jimcarst — APPROVED** [review]
ok for nl

**Review by NeatNit — APPROVED** [review]
`he` looks good as well
@Jean-BaptisteC I'm new here :sweat_smile: how should I respond to a request for review from translations? Do you want a "looks okay" reply from each language, or should I only reply if it's wrong to avoid spamming? And is it fine to approve changes, as @jimcarst and I just did, on behalf of all translation teams despite only checking one language each?

File: data/strings/strings.txt [hunk 2]
```diff
@@ -26937,3 +26937,47 @@
...
+    id = Browser web tidak tersedia
+    it = Il browser web non è disponibile
+    ja = ウェブブラウザが利用できない
+    ko = 웹 브라우저를 사용할 수 없습니다.
+    lt = Interneto naršyklė nepasiekiama
+    mr = वेब ब्राउझर उपलब्ध नाही
+    nb = Nettleser er ikke tilgjengelig
+    nl = Webbrowser is niet beschikbaar
+    pl = Przeglądarka internetowa jest niedostępna
+    pt = O navegador da Web não está disponível
```
Comment (matheusgomesms) [inline data/strings/strings.txt:26972]:
```suggestion
    pt = O navegador de Internet não está disponível
    pt-BR = O navegador da Web não está disponível
```

**Review by strump — APPROVED** [review]

**Review by d4f5409d — APPROVED** [review]
Hu seems right

File: data/strings/strings.txt (hunk 2 above)
Comment (Jean-BaptisteC) [reply data/strings/strings.txt:26972]: Done

File: data/strings/strings.txt [hunk 3]
```diff
@@ -26937,3 +26937,48 @@
     vi = Đi bộ đường dài
     zh-Hans = 徒步旅行
     zh-Hant = 遠足
+
+  [browser_not_available]
+    comment = Text displayed when we call browser and device has no browser
```
Comment (biodranik) [inline data/strings/strings.txt:26942]:
```suggestion
    comment = Text displayed when a webview or a browser are disabled, and external links can't be opened.
```

**Review by biodranik — COMMENTED** [review]
Some checks have failed due to a missing strings regeneration commit?

**Comment by Jean-BaptisteC** [comment]
Ready to merge

**Review by vng — APPROVED** [review]

**Pull request merged**